lib/LeagueUtil.js

```diff
@@ -0,0 +1,294 @@
+const deepmerge = require('deepmerge');
+const fs = require('fs');
+const path = require('path');
+
+const ParameterError = require('./errors/ParameterError');
+
+const RateLimiter = require('./RateLimiter');
+
+class LeagueUtil {
+	// TODO: maybe split into more specific classes instead of "Util" class where everything gets dumped into
+	static normalizeSummonerName(name) {
+		return name.toLowerCase().replace(/ /g, '');
+	}
+
+	/**
+	 * @return null if the name is valid, else the matching characters that are invalid*/
+	static validateSummonerNameInputCharacters(name) { // TODO: add platformID based character validation
+		// validating only allowed characters
+		//  https://developer.riotgames.com/getting-started.html
+		// regexp matches all unicode and ASCII letters and allowed other chars (space, dot and underscore)
+		// needs testing for the unicode characters
+
+		// NA/OCE:
+		// 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz
+		// EUW:
+		// 	0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzªµºÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿĄąĆćĘęıŁłŃńŒœŚśŠšŸŹźŻżŽžƒˆˇˉμﬁﬂ
+		// EUNE:
+		// 	0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzªµºÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿĂăĄąĆćĘęıŁłŃńŐőŒœŚśŞşŠšŢţŰűŸŹźŻżŽžƒȘșȚțˆˇˉΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩάέήίαβγδεζηθικλμνξοπρςστυφχψωόύώﬁﬂ
+		// Brazil:
+		// 	0123456789 ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzÀÁÂÃÇÉÊÍÓÔÕÚàáâãçéêíóôõú
+		// Russia:
+		// 	0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzАБВГДЕЁЖЗИЙКЛМНОПРСТУФХЦЧШЩЪЫЬЭЮЯабвгдеёжзийклмнопрстуфхцчшщъыьэюя
+		// Turkey:
+		// 	ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïð 0123456789ABCDEFGĞHIİJKLMNOPQRSŞTUVWXYZabcçdefgğhıijklmnoöpqrsştuvwxyzªµºÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿıŁłŒœŠšŸŽžƒˆˇˉμﬁﬂĄąĘęÓóĆćŁłŃńŚśŹźŻż
+		// LATAM:
+		// 	0123456789ABCDEFGHIJKLMNÑOPQRSTUVWXYZÁÉÍÓÚÜ abcdefghijklmnñopqrstuvwxyzáéíóúü
+
+		return name.match(/[^\d\u00BF-\u1FFF\u2C00-\uD7FF\w _\.]+/g);
```

Comment:
From the [getting started page](https://developer.riotgames.com/getting-started.html), Riot says this one universally matches (`^[0-9\\p{L} _\\.]+$`). Is there a reason why this one is more specific?